airflow/cli/commands/celery_command.py

```diff
@@ -23,8 +23,10 @@
 from typing import Optional
 
 import daemon
+import psutil
 from celery.bin import worker as worker_bin
 from daemon.pidfile import TimeoutPIDLockFile
```

Comment:
Seems not used any more - can we still use it in our implementation ?

Reply:
Hm, and there's no pylint error?

In my opinion we should use Celery builtin mechanism:
https://docs.celeryproject.org/en/latest/reference/celery.bin.worker.html#cmdoption-celery-worker-pidfile
the main advantage is that in this way it works for both worker's modes.

Reply:
Agree. I looked at what celery uses and it's also PidFile under the hood + atexit removal of the lock, so I am perfectly OK.
 https://github.com/celery/celery/blob/71d024446bbe3e0c9cdffe63fda2c83db42c7fc3/celery/platforms.py#L258

Reply:
And let's check why we have no pylint/flake warning :) ... Maybe the file is in the TO

Reply:
Ouh, it's used in flower command 😺